pkg/types/types.go

```diff
@@ -76,3 +78,33 @@ type MercuryProvider interface {
 	OnchainConfigCodec() mercury.OnchainConfigCodec
 	ContractTransmitter() mercury.Transmitter
 }
+
+type FunctionsProvider interface {
+	PluginProvider
+	LogPollerWrapper() LogPollerWrapper
+}
+
+type OracleRequest struct {
+	RequestID           [32]byte
+	RequestingContract  ocrtypes.Account
+	RequestInitiator    ocrtypes.Account
+	SubscriptionId      uint64
+	SubscriptionOwner   ocrtypes.Account
+	Data                []byte
+	DataVersion         uint16
+	Flags               [32]byte
+	CallbackGasLimit    uint64
+	TxHash              []byte
+	CoordinatorContract ocrtypes.Account
+	OnchainMetadata     []byte
+}
+
+type OracleResponse struct {
+	RequestID [32]byte
+}
+
+// A LogPoller wrapper that understands router proxy contracts
+type LogPollerWrapper interface {
```

Comment:
WDYT about calling this something like `OracleEvents` or `FunctionsContract`?
```suggestion
type OracleEvents interface {
```
```suggestion
type FunctionsContract interface {
```

Reply:
You're right that LogPollerWrapper is a bad name but I also think it might do more things than just supplying Oracle Events in the future, let me thing about it.

Reply:
Maybe FunctionsEvents...